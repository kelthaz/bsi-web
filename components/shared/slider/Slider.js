import PropTypes from 'prop-types';
import styles from './slider.module.scss';

const Slider = ({ name, value, onChange, onBlur, min, max, step }) => {
  const hundredPercent = max - min;

  return (
    <div className={styles['slide-container']}>
      <input
        name={name}
        type="range"
        min={min}
        max={max}
        step={step}
        value={value}
        onChange={onChange}
        onBlur={onBlur}
        style={{
          background: `linear-gradient(to right, #81c1ea ${(((value - min) / hundredPercent) * 100) / 3}% , #338cbf ${
            ((((value - min) / hundredPercent) * 100) / 3) * 2
          }%, #225aa7 ${((value - min) / hundredPercent) * 100}%, #dae0e9 ${((value - min) / hundredPercent) * 100}%`,
        }}
        className={styles.slider}
      />
    </div>
  );
};

Slider.propTypes = {
  name: PropTypes.string.isRequired,
  value: PropTypes.number.isRequired,
  onChange: PropTypes.func.isRequired,
  onBlur: PropTypes.func.isRequired,
  min: PropTypes.number.isRequired,
  max: PropTypes.number.isRequired,
  step: PropTypes.number.isRequired,
};

export default Slider;
