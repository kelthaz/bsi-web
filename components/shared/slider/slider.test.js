import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import React from 'react';
import Slider from './Slider';

describe('Pruebas en el componente Accordion', () => {
  const value = 650000;
  const onChange = jest.fn();
  const onBlur = jest.fn();

  test('Debe de mostrarse correctamente el componente', () => {
    // arrange
    const props = { name: 'monto', value, onChange, onBlur, min: 0, max: 10, step: 1 };
    // act
    const wrapper = shallow(<Slider {...props} />);
    // assert
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe de llamarse el setFieldValue con el valor nuevo y campo a cambiar', () => {
    // arrange
    const event = { target: { value: 100000 } };
    const props = { name: 'monto', value, onChange, onBlur, min: 0, max: 10, step: 1 };
    const wrapper = shallow(<Slider {...props} />);
    // act
    wrapper.find('input').simulate('change', event);
    // assert
    expect(onChange).toHaveBeenCalledTimes(1);
    expect(onChange).toHaveBeenCalledWith(event);
  });
});
