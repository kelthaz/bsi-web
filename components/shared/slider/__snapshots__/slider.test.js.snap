// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente Accordion Debe de mostrarse correctamente el componente 1`] = `
<div
  className="slide-container"
>
  <input
    className="slider"
    max={10}
    min={0}
    name="monto"
    onBlur={[MockFunction]}
    onChange={[MockFunction]}
    step={1}
    style={
      Object {
        "background": "linear-gradient(to right, #81c1ea 2166666.6666666665% , #338cbf 4333333.333333333%, #225aa7 6500000%, #dae0e9 6500000%",
      }
    }
    type="range"
    value={650000}
  />
</div>
`;
